### PR TITLE
Add test code for sending delete and backspace keys

### PR DESCRIPTION
**Pull request by jplandry**
## Change list

Added a test for sending delete keys to a text edit
Added a test for sending backspace keys to a text edit
 
## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Details

Only added tests.

### CODE REVIEW

**Comment by simongranger** [comment]
make sure you run `mvn clean site`
https://github.com/appium/java-client/blob/master/docs/Note-for-developers.md#code-style

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java [hunk 1]
```diff
@@ -242,6 +242,51 @@ public void findToggleButtonAndToggleSeveralTimesTest() throws Exception {
...
+        WebElement field = driver.findElement(By.name("TextEdit"));
+        field.sendKeys(sentText);
+        utils.delayInSeconds(2);
+        Assert.assertEquals(sentText, field.getText());
+
+        for (int i = 0; i < deletedText.length(); ++i) {
+            field.sendKeys(Keys.BACK_SPACE);
+        }
+        Assert.assertEquals(expectedText, field.getText());
+    }
```
Comment (HeloiseRD) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: What happens if you: send BACK_SPACE to an empty field? Send keys to a field that is not a Text Edit field?

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java [hunk 2]
```diff
@@ -242,6 +242,51 @@ public void findToggleButtonAndToggleSeveralTimesTest() throws Exception {
...
+        String deletedText = " Engine";
+        String sentText = expectedText + deletedText;
+
+        WebElement field = driver.findElement(By.name("TextEdit"));
+        field.sendKeys(sentText);
+        utils.delayInSeconds(2);
+        Assert.assertEquals(sentText, field.getText());
+
+        // move the cursor to the front of the text to be deleted
+        for (int i = 0; i < deletedText.length(); ++i) {
```
Comment (HeloiseRD) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: What happens if the cursor is at the end of the text?

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java [hunk 3]
```diff
@@ -242,6 +242,51 @@ public void findToggleButtonAndToggleSeveralTimesTest() throws Exception {
...
+            field.sendKeys(Keys.ARROW_LEFT);
+        }
+        utils.delayInSeconds(2);
+        Assert.assertEquals(sentText, field.getText());
+
+        for (int i = 0; i < deletedText.length(); ++i) {
+            field.sendKeys(Keys.DELETE);
+        }
```
Comment (HeloiseRD) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: What happens if we send DELETE to an empty text field?

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java (hunk 1 above)
Comment (pfoster-youitv) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: not sure if that's a test for this or for how uswish handles the backspace key when it's empty?

The send keys to a field that isn't a text edit could be an interesting test - like enter or space for the togglebutton. I suspect it would behave according to the device (desktop would toggle, but we're not talking desktop here). Definitely good to know.

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java [hunk 4]
```diff
@@ -242,6 +242,51 @@ public void findToggleButtonAndToggleSeveralTimesTest() throws Exception {
...
+
+        WebElement field = driver.findElement(By.name("TextEdit"));
+        field.sendKeys(sentText);
+        utils.delayInSeconds(2);
+        Assert.assertEquals(sentText, field.getText());
+
+        // move the cursor to the front of the text to be deleted
+        for (int i = 0; i < deletedText.length(); ++i) {
+            field.sendKeys(Keys.ARROW_LEFT);
+        }
```
Comment (pfoster-youitv) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: Could combine this to save some loops. 

field.sendKeys(Keys.ARROW_LEFT);
field.sendKeys(Keys.DELETE);

you'd lose the check that the arrow key ends up deleting...

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java (hunk 3 above)
Comment (pfoster-youitv) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: I'm not sure if that's something we need to test here because it's more of a 'how does uswish handled a delete key when the field is empty' kind of thing and less of a 'does the delete key work' test. Up to you guys.

**Comment by pfoster-youitv** [comment]
Does this include the AEP changes you mentioned earlier? About moving some of the views off to a different page?

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java (hunk 2 above)
Comment (jplandry) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: Nothing, this is a feature of CYITextEditView.

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java (hunk 1 above)
Comment (jplandry) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: Nothing happens, this is a feature of CYITextEditView.

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java (hunk 3 above)
Comment (jplandry) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: Nothing happens, this is a feature of CYITextEditView.

File: src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java (hunk 1 above)
Comment (jplandry) [inline src/test/java/io/appium/java_client/YouiEngine/YouiEngineAppiumSampleTest.java]: Sending events to views that do not handle them also does nothing.

**Comment by jplandry** [comment]
```mvn clean site``` 
BUILD SUCCESS

**Comment by HeloiseRD** [comment]
+1
